lrlex/src/lib/lexer.rs

```diff
@@ -40,47 +45,155 @@ pub struct LexFlags {
+                        kind: HeaderErrorKind::ConversionError("LexFlags", "Expected numeric"),
+                        locations: vec![loc.clone()],
+                    })?,
+                    None => None,
+                }
+            };
         }
-        if other.dfa_size_limit.is_some() {
-            self.dfa_size_limit = other.dfa_size_limit;
+        cvt_num!(size_limit, usize);
+        cvt_num!(dfa_size_limit, usize);
+        cvt_num!(nest_limit, u32);
+        Ok(lex_flags)
+    }
+}
+
+impl From<&LexFlags> for Header<Location> {
+    fn from(flags: &LexFlags) -> Header<Location> {
+        let mut header = Header::new();
+        let LexFlags {
+            dot_matches_new_line,
+            multi_line,
+            octal,
+            posix_escapes,
+            allow_wholeline_comments,
+            case_insensitive,
+            swap_greed,
+            ignore_whitespace,
+            unicode,
+            size_limit,
+            dfa_size_limit,
+            nest_limit,
+        } = flags;
+        macro_rules! cvt_flag {
+            ($it: ident) => {
+                $it.map(|x| {
+                    header.insert(
+                        stringify!($it).to_string(),
+                        HeaderValue(
+                            Location::Other("From<&LexFlags".to_string()),
+                            Value::Flag(x, Location::Other("From<&LexFlags>".to_string())),
+                        ),
+                    )
+                });
+            };
         }
-        if other.nest_limit.is_some() {
-            self.nest_limit = other.nest_limit;
+        cvt_flag!(dot_matches_new_line);
+        cvt_flag!(multi_line);
+        cvt_flag!(octal);
+        cvt_flag!(posix_escapes);
+        cvt_flag!(allow_wholeline_comments);
+        cvt_flag!(case_insensitive);
+        cvt_flag!(swap_greed);
+        cvt_flag!(ignore_whitespace);
+        cvt_flag!(unicode);
+
+        macro_rules! cvt_num {
+            ($it: ident) => {
+                $it.map(|x| {
+                    use cfgrammar::header::Setting;
+                    header.insert(
+                        stringify!($it).to_string(),
+                        HeaderValue(
+                            Location::Other("From<&LexFlags".to_string()),
+                            Value::Setting(Setting::Num(
+                                x as u64,
+                                Location::Other("From<&LexFlags>".to_string()),
+                            )),
+                        ),
+                    )
+                });
+            };
         }
+        cvt_num!(size_limit);
+        cvt_num!(dfa_size_limit);
+        cvt_num!(nest_limit);
+
+        header
+    }
+}
+
+pub(crate) fn default_lex_flags_header() -> Header<Span> {
+    // Because this is called in `from_str`, which needs a span to convert to `LexBuildError`
+    // there doesn't seem to be any way to avoid making up this span out of thin air.
```

Comment:
So, One thing we could do here is just remove this function and implement the defaults manually on `LexFlags` so we don't have to synthesize `Span` like this.

It unfortunately gets called from a few places, so perhaps `pub(crate) fn setup_default_lex_flags(&mut LexFlags)`.

Reply:
The best I could think of is in [82069cc](https://github.com/softdevteam/grmtools/pull/556/commits/82069ccbbd9787c5d96209024dbfeb47bcae00a1)

We couldn't do this so late as when constructing rules, because they affect parsing of the `.l` file.
Also I didn't document these shouldn't be marked required, since the constructors already document their
default values it seemed redundant.